django/thunderstore/repository/models/team.py

```diff
@@ -245,39 +246,51 @@ def ensure_user_can_access(self, user: Optional[UserType]) -> None:
         if not self.get_membership_for_user(user):
             raise ValidationError("Must be a member to access team")
 
-    def ensure_can_upload_package(self, user: Optional[UserType]) -> None:
+    def ensure_can_upload_package(
+        self,
+        user: Optional[UserType],
+        namespace: Optional[Namespace] = None,
+        check_namespace: bool = True,
+    ) -> None:
         if not user or not user.is_authenticated:
             raise ValidationError("Must be authenticated")
         if not user.is_active:
             raise ValidationError("User has been deactivated")
         membership = self.get_membership_for_user(user)
         if not membership:
-            raise ValidationError("Must be a member of identity to upload package")
+            raise ValidationError("Must be a member of team to upload package")
         if not self.is_active:
             raise ValidationError(
                 "The team has been deactivated and as such cannot receive new packages"
             )
+        if (
```

Comment:
This check should probably be handled by the namespace itself instead of the team doing it for the namespace.